californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -397,16 +396,17 @@ public synchronized void start() {
 	@Override
 	public synchronized void stop() {
 		if (running) {
+			running = false;
+			for (Exchange exchange : exchangesByMID.values()) {
+				exchange.getRequest().setCanceled(true);
+			}
```

Comment:
Because only the MID exchanges will report timeouts.
The exchanges only referred by tokens are already ACKed, so their maintenance is left to the caller. That's what I assume.

Reply:
Personally, I was expected that the exchange store was "totally clear" OR "not clear at all".

Reply:
FMPOV, there is a difference between `clear` and "callback fro open requests".

All stores are cleared:
```
deduplicator.stop();
exchangesByMID.clear();
exchangesByToken.clear();
```

But only some exchanges are "open requests", and therefore, I only call the cancel for MID exchanges. But, yes, If someone else is able to spent time in start/stop tests, I'm pretty sure, this will also show some surprises as it did for UPDConnector (PR #710) or the DTLSConnector (PR #713). 
I'm closer to close this PR unmerged than in investing again days in testing a start/stop with this PR unmerged.